utils/cp_utils.py

```diff
@@ -66,9 +66,8 @@ def run_cellprofiler(
         # A log file is created for each plate or data set name (based on folder name with images)
         with open(pathlib.Path(f"logs/cellprofiler_output_{pathlib.Path(path_to_images).name}.log"), "w") as cellprofiler_output_file:
             # run CellProfiler for a illumination correction pipeline
-            command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}"
-            subprocess.run(command, shell=True, stdout=cellprofiler_output_file, stderr=cellprofiler_output_file)
-        cellprofiler_output_file.close()
+            command = f"cellprofiler -c -r -p {path_to_pipeline} -o {path_to_output} -i {path_to_images}".split()
+            subprocess.run(command, capture_output=cellprofiler_output_file)
```

Comment:
What would happen if this (and the other) subprocess procedure ran into an error?

Reply:
It should stop the process and provide the error in the log file.

Reply:
@jenna-tomkinson - Are you certain this would stop the process if an error was encountered? Consider the following example:

```python
import pathlib
import subprocess

with open(pathlib.Path("example.log"), "w") as logfile:
    subprocess.run(["ls", "-z"], capture_output=logfile)
```

Reply:
I just tested it, and it would move onto the next plate instead. I gave it a path to a non-existent directory and it gave a blank log file and moved on.

I could check to make sure all paths lead to directories and if they don't then raise an error?

Would that suffice? I know that if there is an error occurring in CellProfiler that it normally stops the run in the log file from my experience but I could be wrong.